docs/00.MIGRATION-GUIDE.md

```diff
@@ -0,0 +1,372 @@
+
+  jerry_value_t generate_result;
+  generate_result = jerry_generate_snapshot (NULL,
+                                             0,
+                                             code_to_snapshot_p,
+                                             strlen ((const char *) code_to_snapshot_p),
+                                             global_mode_snapshot_buffer,
+                                             sizeof (global_mode_snapshot_buffer) / sizeof (uint32_t));
+  size_t snapshot_size = (size_t) jerry_get_number_value (generate_result);
+  jerry_release_value (generate_result);
+  jerry_cleanup ();
+}
+```
+
+### jerry_parse_and_save_function_snapshot
+
+This function was replaced with [`jerry_generate_function_snapshot`](02.API-REFERENCE.md#jerry_parse_and_save_function_snapshot).
+
+**Before**
+
+```c
+{
+  const jerry_char_t *args_p = (const jerry_char_t *) "a, b";
+  const jerry_char_t *src_p = (const jerry_char_t *) "return a + b;";
+
+  size_t func_snapshot_size =
+  jerry_parse_and_save_function_snapshot (src_p,
+                                          strlen ((const char *) src_p),
+                                          args_p,
+                                          strlen ((const char *) args_p),
+                                          false,
+                                          func_snapshot_buffer,
+                                          sizeof (func_snapshot_buffer) / sizeof (uint32_t));
+
+  jerry_cleanup ();
+}
+```
+
+**After**
+
+```c
+{
+  const jerry_char_t *args_p = (const jerry_char_t *) "a, b";
+  const jerry_char_t *src_p = (const jerry_char_t *) "return a + b;";
+
+  jerry_value_t generate_result;
+  generate_result = jerry_generate_function_snapshot (NULL,
+                                                      0,
+                                                      src_p,
+                                                      strlen ((const char *) src_p),
+                                                      args_p,
+                                                      strlen ((const char *) args_p),
+                                                      0,
+                                                      func_snapshot_buffer,
+                                                      sizeof (func_snapshot_buffer) / sizeof (uint32_t));
+
+  size_t snapshot_size = (size_t) jerry_get_number_value (generate_result);
+  jerry_release_value (generate_result)
+  jerry_cleanup ();
+}
+```
+
+## Removed/renamed
+
+***Removed legacy headers***
+
+- `jerry-api.h`
+- `jerry-port.h`
+- `jerry-internal.h`
+
+***Renamed headers***
+
+- `jerry-port.h` to `jerrysctipt-port.h`
+
+***Removed API functions***
+
+- `jerry_get_memory_limits`
+
+***Removed unused configuration macros***
+
+- `CONFIG_MEM_DATA_LIMIT_MINUS_HEAP_SIZE`
+- `CONFIG_MEM_STACK_LIMIT`
+- `CONFIG_VM_STACK_FRAME_INLINED_VALUES_NUMBER`
+- `CONFIG_ECMA_GLOBAL_ENVIRONMENT_DECLARATIVE`
+
+# New API functions
```

Comment:
What about #2385? The `jerry_gc` function has been extended with an operation mode parameter. It should be mentioned in a`Modified API functions` section.